app/src/main/java/space/taran/arkmemo/ui/fragments/TextNotesFragment.kt

```diff
@@ -19,55 +19,58 @@ import kotlinx.coroutines.launch
-import space.taran.arkmemo.time.MemoCalendar
+import space.taran.arkmemo.models.TextNote
+import space.taran.arkmemo.ui.activities.MainActivity
 import space.taran.arkmemo.ui.activities.getTextFromClipBoard
 import space.taran.arkmemo.ui.activities.replaceFragment
 import space.taran.arkmemo.ui.adapters.TextNotesListAdapter
 
 @AndroidEntryPoint
-class TextNotes: Fragment(R.layout.fragment_text_notes) {
+class TextNotesFragment: Fragment(R.layout.fragment_text_notes) {
 
     private val binding by viewBinding(FragmentTextNotesBinding::bind)
-    private val activity: AppCompatActivity by lazy {
-        requireActivity() as AppCompatActivity
+
+    private val activity: MainActivity by lazy {
+        requireActivity() as MainActivity
     }
-    private val textNotesViewModel: TextNotesViewModel by activityViewModels()
+    private val textNotesViewModel: TextNotesViewModel by viewModels()
 
     private lateinit var newNoteButton: FloatingActionButton
     private lateinit var pasteNoteButton: Button
 
     private lateinit var recyclerView: RecyclerView
 
     private val newNoteClickListener = View.OnClickListener{
-        val editTextNotes = EditTextNotes()
-        (requireActivity() as AppCompatActivity).replaceFragment(editTextNotes, EditTextNotes.TAG)
+        activity.fragment = EditTextNotesFragment()
+        activity.replaceFragment(activity.fragment, EditTextNotesFragment.TAG)
     }
 
     private val pasteNoteClickListener = View.OnClickListener {
         val clipBoardText = requireContext().getTextFromClipBoard()
         if (clipBoardText != null) {
-            val editTextNotes = EditTextNotes(clipBoardText)
-            activity.replaceFragment(editTextNotes, EditTextNotes.TAG)
+            activity.fragment = EditTextNotesFragment.newInstance(clipBoardText)
+            activity.replaceFragment(activity.fragment, EditTextNotesFragment.TAG)
         }
         else Toast.makeText(requireContext(), getString(R.string.nothing_to_paste), Toast.LENGTH_SHORT).show()
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        if (savedInstanceState == null) {
-            recyclerView = binding.include.recyclerView
-            activity.title = getString(R.string.app_name_debug) //Activity title to change before production
-            activity.supportActionBar?.setDisplayHomeAsUpEnabled(false)
-            newNoteButton = binding.newNote
-            pasteNoteButton = binding.pasteNote
-            newNoteButton.setOnClickListener(newNoteClickListener)
-            pasteNoteButton.setOnClickListener(pasteNoteClickListener)
-            lifecycleScope.launch {
-                viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+        recyclerView = binding.include.recyclerView
+        activity.title = getString(R.string.app_name_debug) //Activity title to change before production
```

Comment:
Still should be changed :) Branch `main` is production.

Reply:
Alright